include/multipass/settings.h

```diff
@@ -41,6 +41,20 @@ class Settings : public Singleton<Settings>
     virtual QString get(const QString& key) const;            // throws on unknown key
     virtual void set(const QString& key, const QString& val); // throws on unknown key or bad settings
 
+    /**
+     * Obtain a setting as a certain type
+     * @tparam T The type to obtain the setting as
+     * @param key The Key that identifies the setting
+     * @return The value that the setting converts to, when interpreted as type @p T. Follows the conversion rules used
+     * by @c QVariant.
+     * If the current @em value cannot be converted, a default @p T value is returned.
+     * @note QVariant's conversion rules may be surprising and not always correctly documented. For instance according
+     * to Qt's documentation at the time of writing, @c "False" would convert to @c bool as @c true, but it actually
+     * converts to @c false. However, @c "no" converts to @c true.
```

Comment:
Actually, I misread the docs. Here's the doc I was referring to: https://doc.qt.io/qt-5/qvariant.html#toBool

> Returns true if the variant has userType() QMetaType::Bool, QMetaType::QChar, QMetaType::Double, QMetaType::Int, QMetaType::LongLong, QMetaType::UInt, or QMetaType::ULongLong and the value is non-zero, or if the variant has type QMetaType::QString or QMetaType::QByteArray and its lower-case content is not one of the following: empty, "0" or "false"; otherwise returns false.

I missed "lower-case" in there. I'll remove the note.